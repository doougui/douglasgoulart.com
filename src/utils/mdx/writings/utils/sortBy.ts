import { Writing } from 'types/Writing';
import { parseToDate } from 'utils/time/parseToDate';
import { SortTypesKeys } from '../types';

function byNew(data: Writing[]) {
  return data.sort((a, b) => {
    const aAsDate = parseToDate(a.publishedAt);
    const bAsDate = parseToDate(b.publishedAt);
    return Number(bAsDate) - Number(aAsDate);
  });
}

function byName(data: Writing[]) {
  return data.sort((a, b) => (a.title > b.title ? 1 : -1));
}

export function sortBy(data: Writing[], type: SortTypesKeys): Writing[] {
  const sortTypes: { [key in SortTypesKeys]: () => Writing[] } = {
    new: () => byNew(data),
    name: () => byName(data),
  };

  return sortTypes[type]();
}
