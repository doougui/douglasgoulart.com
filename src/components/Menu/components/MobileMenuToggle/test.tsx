import { render } from 'utils/tests';

import { MobileMenuToggle } from '.';

describe('<MobileMenuToggle />', () => {
  it('should render correctly', () => {
    const { container } = render(<MobileMenuToggle />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
