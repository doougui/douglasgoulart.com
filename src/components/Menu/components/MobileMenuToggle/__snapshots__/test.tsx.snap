// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MobileMenuToggle /> should render correctly 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 4px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 {
  all: unset;
}

.c2 {
  width: 2.7rem;
  height: 3px;
  border-radius: 2px;
  background-color: white;
}

@media (min-width:768px) {
  .c1 {
    display: none;
  }
}

<button
  class="c0"
  type="button"
>
  <nav
    class="c1"
  >
    <span
      class="c2"
    />
    <span
      class="c2"
    />
  </nav>
</button>
`;
